### PR TITLE
generators/kzg_4844: Add positive G1_POINT_AT_INFINITY test vectors

### PR DESCRIPTION
**Pull request by asn-d6**
In a similar vein to https://github.com/ethereum/consensus-specs/pull/3507, this PR is adding some **valid** test vectors for valid edge-case blobs whose proof or commitment is the point at infinity.

In particular:
- Proof being the point at infinity is a valid case for when the blob is all zeroes. A blob being all zeroes is basically the zero polynomial -- its commitment will be the point at infinity and the corresponding proof will also be the point at infinity.
- A related edge case can happen for a blob full of twos (or any other constant number): A blob full of twos would correspond to the polynomial p(x) = 2 and it's corresponding KZG proof would be the point at infinity. It's corresponding commitment would be 2G (not the point at infinity).

As part of this commit, I also edited the `verify_kzg_proof_case_incorrect_proof_point_at_infinity()` test to loop over all valid field elements for the evaluation point (lmk if you want me to split it to another commit).

Also, I didn't modify the `verify_blob_kzg_proof_batch()` test vectors because the edge-case will be tested automatically by adding `BLOB_ALL_TWOS` to `VALID_BLOBS` and the test `verify_blob_kzg_proof_batch_case_*()`.

/cc @jtraglia

### CODE REVIEW

File: tests/generators/kzg_4844/main.py [hunk 1]
```diff
@@ -384,6 +424,38 @@ def case05_verify_blob_kzg_proof():
         'output': False
     }
 
+    # Correct `G1_POINT_AT_INFINITY` proof and commitment for zero poly
+    blob = BLOB_ALL_ZEROS
```
Comment (hwwhww) [inline tests/generators/kzg_4844/main.py:431]: Just note that this test case is duplicated to the first test case in `verify_blob_kzg_proof_case_correct_proof_*` test cases. But I think it's good to make it explicit.

**Review by hwwhww — APPROVED** [review]
I fixed the minor linter errors.

It's great to add `BLOB_ALL_TWOS` test cases! 👍


@jtraglia 
The `kzg` test vectors with this PR: [kzg.tar.gz](https://github.com/ethereum/consensus-specs/files/12713750/kzg.tar.gz)

Could you help test it against c-kzg before we merge it? 🙏

File: tests/generators/kzg_4844/main.py (hunk 1 above)
Comment (asn-d6) [reply tests/generators/kzg_4844/main.py:431]: Yep, indeed. I also considered it being explicit a positive thing.

**Review by jtraglia — APPROVED** [review]
LGTM 👍 Can confirm these tests pass in c-kzg. Thanks @asn-d6!
